### PR TITLE
Report unknown tasks

### PR DESCRIPTION
**Pull request by asafchen-dig**

### CODE REVIEW

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt [hunk 1]
```diff
@@ -6,3 +6,67 @@ const val OTEL_AGENT_JAR_URL = "https://github.com/open-telemetry/opentelemetry-
...
+     * Gradle -> build setup
+     */
+    "init",
+    "wrapper",
+
+    /*
+     * Gradle -> documentation
+     */
+    "javadoc",
+
```
Comment (arik-dig) [inline java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt:50]: I think you're missing Gradle -> other

```
compileJava
compileKotlin
compileTestJava
compileTestKotlin
```

and maybe more

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt [hunk 2]
```diff
@@ -6,3 +6,67 @@ const val OTEL_AGENT_JAR_URL = "https://github.com/open-telemetry/opentelemetry-
 const val DIGMA_AGENT_EXTENSION_JAR_URL = "https://github.com/digma-ai/otel-java-instrumentation/releases/latest/download/digma-otel-agent-extension.jar"
 const val OTEL_AGENT_JAR_NAME = "opentelemetry-javaagent.jar"
 const val DIGMA_AGENT_EXTENSION_JAR_NAME = "digma-otel-agent-extension.jar"
+
+val UNIMPORTANT_TASKS = listOf(
```
Comment (arik-dig) [inline java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt:10]:
```suggestion
val KNOWN_IRRELEVANT_TASKS = listOf(
```

Comment (arik-dig) [inline java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt:10]: I think need different lists for Maven and Gradle.
and maybe we can exclude groups , not sure if you seen via debug but for example test would be :test:abc , so there's this : thing

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt [hunk 3]
```diff
@@ -6,3 +6,67 @@ const val OTEL_AGENT_JAR_URL = "https://github.com/open-telemetry/opentelemetry-
...
+    "projects",
+    "properties",
+    "tasks",
+
+    /*
+     * Gradle -> verification
+     */
+    "check",
+    "test",
+
```
Comment (arik-dig) [inline java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt:71]: Quarkus (maven)

can exclude all `:quarkus:*` expect 
**quarkus:dev
quarkus:remote-dev**

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt [hunk 4]
```diff
@@ -178,6 +179,31 @@ class ActivityMonitor(private val project: Project) /*: Runnable, Disposable*/ {
         )
     }
 
+    fun reportUnknownTaskRunning(configType: String, taskNames: List<String>) {
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt:182]:
```suggestion
    fun reportUnknownTaskRunning(buildSystem: String, taskNames: List<String>) {
```

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt [hunk 5]
```diff
@@ -178,6 +179,31 @@ class ActivityMonitor(private val project: Project) /*: Runnable, Disposable*/ {
...
+            .filter{ !latestUnknownRunConfigTasks.containsKey(it)}
+            .toList()
+        for (task in taskNames)
+            latestUnknownRunConfigTasks[task] = Instant.now()
+
+        if(taskNamesToReport.isEmpty())
+            return
+
+        postHog?.capture(
+            userId, "unknown-config ran", mapOf(
```
Comment (arik-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt:200]:
```suggestion
            userId, "unknown-run-config", mapOf(
```

**Review by arik-dig — COMMENTED** [review]
2nd round

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt (hunk 1 above)
Comment (asafchen-dig) [reply java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt:50]: added

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt (hunk 2 above)
Comment (asafchen-dig) [reply java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/constants.kt:10]: renamed

File: java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/RunCfgTools.kt [hunk 6]
```diff
@@ -54,6 +55,22 @@ class RunCfgTools {
...
         }
 
+        @JvmStatic
+        fun reportUnknownTasksToPosthog(configuration: RunConfigurationBase<*>){
+            val activityMonitor = ActivityMonitor.getInstance(configuration.project)
+
+            if (configuration is GradleRunConfiguration) {
+                val taskNames = configuration.settings.taskNames
+                taskNames.removeAll(UNIMPORTANT_TASKS)
+                activityMonitor.reportUnknownTaskRunning("gradle", taskNames)
```
[the remark this answers is not in the document]
Comment (asafchen-dig) [reply java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/RunCfgTools.kt:65]: added

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt (hunk 4 above)
Comment (asafchen-dig) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt:182]: cahnged

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt (hunk 5 above)
Comment (asafchen-dig) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/posthog/ActivityMonitor.kt:200]: no

**Pull request closed without merging**